community/ndp/runtime/src/test/java/org/neo4j/ndp/runtime/integration/RecordingCallback.java

```diff
@@ -219,17 +221,17 @@ public String toString()
         }
     }
 
-    private Object[] unwind( RecordStream result ) throws Exception
+    private Record[] unwind( RecordStream result ) throws Exception
     {
-        final List<Object> values = new ArrayList<>();
+        final List<Record> values = new ArrayList<>();
         result.visitAll( new RecordStream.Visitor()
         {
             @Override
             public void visit( Record record )
             {
-                values.add( record.copy() );
+                values.add( new ImmutableRecord( record.fields() ) );
```

Comment:
This works fine. We may need to come back to this later on if we start passing these records around outside their transactions - but for now there is indeed no need for the copy method outside of tests. +1 to removing it from production code.

This needs to be rebased now though, since I merged two of the other ones.